### ISSUE TITLE
About the "self.stem = nn.Conv3d(in_channels, n_channels, kernel_size=1)"

### ISSUE THREAD

**Issue by islandLZ** [MIC-DKFZ/MedNeXt#9]
The data format read by my data loader is torch.Size([1, 3, 512, 512]).
![image](https://github.com/MIC-DKFZ/MedNeXt/assets/65443889/cd4b6686-86af-432c-b18a-c05c182ca389)

I encountered an error here。
![image](https://github.com/MIC-DKFZ/MedNeXt/assets/65443889/71e10f7c-813f-495d-9d9a-695f5bae3210)

So how should I set the initial parameters of the entire network?
![image](https://github.com/MIC-DKFZ/MedNeXt/assets/65443889/881aafc1-5973-4ab1-a87d-848615a74145)

When I put in_ A new error occurred when in_channels were set to 1：
![image](https://github.com/MIC-DKFZ/MedNeXt/assets/65443889/15da33c6-0794-4795-a655-503d91ad06cb)

**Comment by islandLZ** [comment]
When I change n_channels = 3 ， I meet an other erro！

**Comment by saikat-roy** [comment]
Hi @islandLZ. The problem seems to be the last version of the architecture was strictly designed to be 3d while your inputs are 2d. I've just committed a version of the architecture with a 'dim' argument with '2d' or '3d' inputs. Give that a try and see if it works for you.

**Comment by islandLZ** [comment]
I set this：

class MedNeXt(nn.Module):

    def __init__(self, 
        in_channels: int = 3, 
        n_channels: int = 32,
        n_classes: int = 3, 
        exp_r: list = [3,4,8,8,8,8,8,4,3],         # Expansion ratio as in Swin Transformers
        kernel_size: int = 5,                      # Ofcourse can test kernel_size
        enc_kernel_size: int = None,
        dec_kernel_size: int = None,
        deep_supervision: bool = False,             # Can be used to test deep supervision
        do_res: bool = True,                       # Can be used to individually test residual connection
        do_res_up_down: bool = True,             # Additional 'res' connection on up and down convs
        checkpoint_style: str = 'outside_block',  # Either inside block or outside block
        block_counts: list = [3,4,8,8,8,8,8,4,3], # Can be used to test staging ratio:# [3,3,9,3] in Swin as opposed to [2,2,2,2,2] in nnUNet
        norm_type = 'group',
    ):
   .......
then I have made modifications to my data format：
[1,3,512,512] -> [1,3,64,64,64]

So that the network can operate normally, I am still testing whether this method is feasible。

**Comment by islandLZ** [comment]
> Hi @islandLZ. The problem seems to be the last version of the architecture was strictly designed to be 3d while your inputs are 2d. I've just committed a version of the architecture with a 'dim' argument with '2d' or '3d' inputs. Give that a try and see if it works for you.

Thank you for your work. I will try the 'dim' you just set up now。

**Comment by islandLZ** [comment]
I success

> Hi @islandLZ. The problem seems to be the last version of the architecture was strictly designed to be 3d while your inputs are 2d. I've just committed a version of the architecture with a 'dim' argument with '2d' or '3d' inputs. Give that a try and see if it works for you.

Your work has been successful.
![image](https://github.com/MIC-DKFZ/MedNeXt/assets/65443889/8d1d7634-a91c-409e-a368-874edac76dde)
![image](https://github.com/MIC-DKFZ/MedNeXt/assets/65443889/17129fd7-3851-4fc1-bc49-2110f22ad60b)

**Comment by saikat-roy** [comment]
I'm glad to hear that it helped!
